javalanglib/src/main/scala/java/lang/StringBuilder.scala

```diff
@@ -65,132 +138,91 @@ class StringBuilder(private var content: String) extends CharSequence
   def lastIndexOf(str: String, fromIndex: Int): Int =
     content.lastIndexOf(str, fromIndex)
 
-  def subSequence(start: Int, end: Int): CharSequence = substring(start, end)
-  def substring(start: Int): String = content.substring(start)
-  def substring(start: Int, end: Int): String = content.substring(start, end)
-
   def reverse(): StringBuilder = {
     val original = content
     var result = ""
-    var i = 0
-    while (i < original.length) {
+    var i = original.length - 1
+    while (i > 0) {
       val c = original.charAt(i)
-      if (Character.isHighSurrogate(c) && (i+1 < original.length)) {
-        val c2 = original.charAt(i+1)
-        if (Character.isLowSurrogate(c2)) {
-          result = c.toString + c2.toString + result
-          i += 2
+      if (Character.isLowSurrogate(c)) {
+        val c2 = original.charAt(i - 1)
+        if (Character.isHighSurrogate(c2)) {
+          result = result + c2.toString + c.toString
+          i -= 2
         } else {
-          result = c.toString + result
-          i += 1
+          result += c.toString
+          i -= 1
         }
       } else {
-        result = c.toString + result
-        i += 1
+        result += c.toString
+        i -= 1
       }
     }
+    if (i == 0)
+      result += original.charAt(0).toString
     content = result
     this
   }
 
-  def deleteCharAt(index: Int): StringBuilder = {
-    if (index < 0 || index >= content.length)
-      throw new StringIndexOutOfBoundsException("String index out of range: " + index)
-    content = content.substring(0, index) + content.substring(index+1)
-    this
-  }
+  override def toString(): String = content
 
-  def ensureCapacity(minimumCapacity: Int): Unit = {
-    // Do nothing
-  }
+  def length(): Int = content.length()
 
-  /**
-   * @param start The beginning index, inclusive.
-   * @param end The ending index, exclusive.
-   * @param str String that will replace previous contents.
-   * @return This StringBuilder.
-   */
-  def replace(start: Int, end: Int, str: String): StringBuilder = {
-    val length = content.length
-    if (start < 0 || start > end || start > length) {
-      throw new StringIndexOutOfBoundsException(
-          s"Illegal to replace substring at [$start - $end] in string of length $length")
-    }
+  def capacity(): Int = length()
 
-    val realEnd = if (end > length) length else end // java api convention
-    content = content.substring(0, start) + str + content.substring(realEnd)
-    this
+  def ensureCapacity(minimumCapacity: Int): Unit = {
+    // Do nothing
```

Comment:
I would just write `def ensureCapacity(...): Unit = ()`. The comment "Do nothing" is not useful. If you want to document better, say **why** we are not doing anything. (But I think it is quite clear).